Change schema table from horizontal to vertical

cafenoctua opened the pull request:
クエリの結果ウィンドウで表示されるスキーマを横から縦に変更してみました。

minodisk commented:
First of all, thank you for sending me the PR 🚀 
The code itself in this PR is good.

I would appreciate it if you could explain in the PR why horizontal table is not good or why vertical table is necessary, which cannot be expressed in the code.
Because after merging this PR, if a PR comes in that says, for example, "Changed the schema table from vertical to horizontal", I will not be able to decide whether I should merge that PR or not.

---

For example, but one reason I can think of is something like, "If we add a description column to the schema in the future, the vertical table should be easier to read".

<img width="1066" alt="image" src="https://user-images.githubusercontent.com/514164/180651559-b01fa1b2-d883-402e-9789-9fccbac86523.png">

cafenoctua commented:
Thank you for your explanation.
The purpose of this addition was to improve visibility even when there were many rows.
Also, as you say, I implemented it because I thought it would be easier to see if I added a description column.

minodisk commented on packages/viewer/src/domain/Schema.tsx:
```diff
@@ -15,27 +15,21 @@ export const Schema: FC<{
+            {heads.map(({ id, type, mode }) => (
+              <Tr><Th key={id}>{id}</Th>
+              <Th key={id}>{type}</Th>
+              <Th key={id}>{mode}</Th></Tr>
```
I think the `key` works when set to the parent element.

```jsx
<Tr key={id}>
  <Th>{id}</Th>
  <Th>{type}</Th>
  <Th>{mode}</Th>
</Tr>
```

minodisk commented on packages/viewer/src/domain/Schema.tsx:
```diff
@@ -15,27 +15,21 @@ export const Schema: FC<{
+              <Th key={id}>{mode}</Th></Tr>
+            ))}
+          </Tr>
+        </Thead>
```
A `THead` is set for a child of `TBody`; the `THead` and `TBody` must be placed in the same hierarchy as children of `Table`:

```jsx
<Tbody>
  <Thead>
  </Thead>
  <TBody>
  </Tbody>
</Table>
```

See [\<table\>: The Table element \- HTML: HyperText Markup Language \| MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/table).

cafenoctua commented:
Thank you for your review
I understood what you meant, so I fixed it.

minodisk commented on packages/viewer/src/domain/Schema.tsx:
```diff
@@ -15,21 +15,23 @@ export const Schema: FC<{
+            <Th>Type</Th>
+            <Th>Mode</Th>
+          </Tr>
+        </Tr>
```
`Tr`s are duplicated. Must be only one.

cafenoctua replied:
Oops
Deleted unnecessary Tr tags.

minodisk reviewed (changes requested):
One more 🏃‍♂️

minodisk reviewed (approved):
LGTM 👍

minodisk commented:
Released as [v1.12.0](https://github.com/minodisk/bigquery-runner/blob/main/CHANGELOG.md#changed) 🎉 

FYI, there was another feature released at the same time that allowed multiple tables handled in child jobs to be displayed as separate table tabs, and the schema tab was merged into the table tab because of that.

![image](https://user-images.githubusercontent.com/514164/182409823-0dd0ea5f-974c-4d8d-b750-c18d09bacf2a.png)
